repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,266 @@
+# SPDX-FileCopyrightText: 2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+
+import base64
+import os
+from hashlib import sha256
+from pathlib import Path
+from typing import Any, Dict, Optional
+from urllib import request
+
+import yaml
+from click import Context
+from tuf.api.exceptions import DownloadError, RepositoryError
+from tuf.ngclient import Updater
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+from repository_service_tuf.cli.artifact.repository import find_hash_bin
+
+DOWNLOAD_DIR = "./downloads"
+
+
+@artifact.command()
+@click.option(
+    "-c",
+    "--config",
+    help="A config file to use.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-r",
+    "--repository",
+    help="A repository to use config for.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-t",
+    "--trusted-root",
+    "trusted_root",
+    help="A metadata URL to the initial trusted root.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-",
+    "--metadata-url",
+    "metadata_url",
+    help="A URL to download TUF metadata from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    "artifacts_url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-n",
+    "--arifact-name",
+    "artifact_name",
+    help="An artifact name to get.",
+    type=bool,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-h",
+    "--hash-prefix",
+    "hash_prefix",
+    help="A hash prefix.",
+    type=bool,
+    required=None,
+    default=False,
+)
+@click.option(
+    "-P",
+    "--directory-prefix",
+    "directory_prefix",
+    help="A prefix for the download dir.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.pass_context
+def download(
+    context: Context,
+    config: str,
+    repository: str,
+    trusted_root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    artifact_name: str,
+    hash_prefix: bool,
+    directory_prefix: str,
+) -> None:
+    """
+    Downloads artifacts to the TUF metadata using python-tuf ``ngclient``.
+    """
+
+    rstuf_config: Dict[str, Any]
+    if config:
+        with open(config, "r") as stream:
+            try:
+                rstuf_config = yaml.safe_load(stream)
+            except yaml.YAMLError as e:
+                console.print(f"Failed to parse config file: {e}")
```

Comment:
We don't need to handle it. 
We use [Dynaconf](https://www.dynaconf.com) and we manage it within config in the `rstuf` command level as I mentioned above.
```suggestion
        rstuf_config = context.obj["settings"]
```